packages/flutter_tools/lib/src/version.dart

```diff
@@ -96,11 +153,141 @@ class FlutterVersion {
 
     return '$branch/$commit';
   }
+
+  /// We warn the user if the age of their Flutter installation is greater than
+  /// this duration.
+  @visibleForTesting
+  static const Duration kVersionAgeConsideredUpToDate = const Duration(days: 4 * 7);
+
+  /// The amount of time we wait before pinging the server to check for the
+  /// availability of a newer version of Flutter.
+  @visibleForTesting
+  static const Duration kCheckAgeConsideredUpToDate = const Duration(days: 1);
```

Comment:
Done. 1 week for `kCheckAgeConsideredUpToDate` and `kVersionAgeConsideredUpToDate = kCheckAgeConsideredUpToDate * 4`.